### DISCUSSION TITLE
Adding Prima Migrate to Project - use my own initial migration?

### DISCUSSION THREAD

**Discussion by kgoggin** [prisma/prisma#6270] [Q&A] [answered]
Our app started life as a Prisma 1 project. We made the migration to Prisma 2 a few months ago but, since Prisma Migrate wasn't GA yet, we decided to wait to set it up and manually migrate things for the time being. As such, we've got a .sql file that we can use to spin up a new DB that matches what we have in prod to be used for local/test environments. Thankfully, we haven't needed to perform any changes to our DB since switching to Prisma 2, so that one file still represents the exact state of our prod DB.

I'm investigating what it'll take to move us to using Prisma Migrate, following the [guide here](https://www.prisma.io/docs/guides/application-lifecycle/add-prisma-migrate-to-a-project). After introspecting our existing DB and running `prisma migrate dev`, I can see a few differences in our existing DB setup vs. what Prisma would generate:
  1. We have a few `updatedAt` fields that are handled natively by the DB (not using `@updatedAt` directive)
  2. Our existing DB has quite a few places where the cascade deletion logic is different than what Prisma generates. My understanding from [this article](https://www.prisma.io/docs/guides/general-guides/database-workflows/cascading-deletes/mysql) is that the right way to handle this would be to manually edit the migration file before it gets applied to change the behavior to what we currently have.

Given the need to make some manual updates to the migration file that Prisma is generating, and given that I'm not actually going to run that migration in my prod DB anyway (I'll be baselining it with `prisma migrate resolved --applied "init"`), I'm thinking it might be easiest for me to just swap out the initial `migration.sql` file that Prisma generates for me with my existing .sql file that already does the job of creating a DB that matches our prod one. My understanding of how everything works is that that'd mean that any newly-created DB would start out exactly the same way we're setting it up currently, because the exact same .sql script would be applied.

The potential issue I see here is if there was something being generated by Prisma as a result of my schema file that didn't match 1:1 with what I've got in my existing .sql script, which would cause things to get out of sync when I started editing my schema file to make a change and generated a new migration. But, as far as I can tell, there are no such differences - all of the differences that exist are things that aren't describable in the schema file (cascade behavior and native update at behavior).

So, my question is: is there any other reason I _shouldn't_ just use my existing setup script as the very first migration.sql file?

**Comment by tomhoule** [comment, marked as answer]
Hi @kgoggin! This is indeed what I would recommend at the moment: if you have a SQL file that exactly matches your production database schema, it should be your first migration file. That way, your development/staging databases created from the migration exactly match your production database. And indeed, if you go through the baselining process, that migration won't run on the prod database.

The prisma schema is indeed "lower resolution" than the database schema — for whatever isn't specified in the prisma schema yet (like index types, check constraints, ON DELETEs....), it's safe to deviate from the migrate default, it shouldn't detect any drift (please open an issue if it does and it will be fixed). FWIW the cascading logic is something we are working on at the moment, so you can expect improvements there in the future.

**Reply by kgoggin** [reply]
Awesome, thanks @tomhoule

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
